src/server/actions/reward/getReward.ts

```diff
@@ -29,30 +29,41 @@ export const getReward = async (id: string) => {
   return reward
 }
 
-export const getRewardPaymentDetails = async (id: string) => {
-  const reward = await db.query.rewards.findFirst({
-    where: (reward, {eq}) => eq(reward.id, id),
-    columns: {
-      amount: true,
-      transferTxHash: true,
-    },
-    with: {
-      user: {
-        columns: {
-          walletAddress: true,
-        },
-      },
-    },
-  })
+export const getRewardPaymentDetails = async (
+  contestId: string,
+  userId: string,
+) => {
+  const reward = await db
+    .select({
+      amount: sum(rewards.amount),
+      userWalletAddress: users.walletAddress,
+    })
+    .from(rewards)
+    .leftJoin(users, eq(rewards.userId, users.id))
+    .leftJoin(findings, eq(rewards.findingId, findings.id))
+    .leftJoin(contests, eq(findings.contestId, contests.id))
+    .groupBy(users.id)
+    .where(
+      and(
+        eq(rewards.userId, userId),
+        isNull(rewards.transferTxHash),
+        eq(contests.id, contestId),
+      ),
+    )
 
-  if (!reward) {
+  const userReward = reward[0]
+
+  if (!userReward) {
     return getApiError('Reward not found')
   }
 
+  if (!userReward.amount) {
+    return getApiError('Reward amount not found')
+  }
```

Comment:
Maybe these could be changed to `throw new ServerError()` and `serializeServerError` so that it's in line with other server actions?